core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -473,10 +473,26 @@ public boolean delete(Object key) {
          resizeLock.readLock().unlock();
       }
 
+      org.infinispan.commons.io.ByteBuffer keyBb;
+      org.infinispan.commons.io.ByteBuffer valueBb = null;
```

Comment:
Ah yes I see.  Interestingly, the invocation you had below didn't need the casts :)  But with my other rework comment it will.